crates/torii/grpc/src/server/mod.rs

```diff
@@ -1042,6 +1043,38 @@ impl DojoWorld {
             .add_subscriber(clause.into_iter().map(|keys| keys.into()).collect())
             .await
     }
+
+    async fn retrieve_controllers(
+        &self,
+        contract_addresses: Vec<Felt>,
+    ) -> Result<proto::world::RetrieveControllersResponse, Error> {
+        let query = if contract_addresses.is_empty() {
+            "SELECT address, username, deployed_at FROM controllers".to_string()
+        } else {
+            format!(
+                "SELECT address, username, deployed_at FROM controllers WHERE address IN ({})",
+                contract_addresses.iter().map(|_| "?".to_string()).collect::<Vec<_>>().join(", ")
+            )
+        };
+
+        let mut db_query = sqlx::query_as::<_, (String, String, u64)>(&query);
```

Comment:
This fails because `deployed_at` is a `TEXT` and not a `u64`. You wanted to return a timestamp via the gRPC? Or it can be a `TEXT` with the date?